src/commands/misc/rename.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  ChatInputCommandInteraction,
+  GuildMemberRoleManager,
+  SlashCommandBuilder,
+} from "discord.js";
+
+import { discordClient } from "../../clients/discord.js";
+import { config } from "../../config.js";
+
+export const data = new SlashCommandBuilder()
+  .setName("rename")
+  .setDescription("Renames our beloved O.A.F.")
+  .addStringOption((o) =>
+    o
+      .setName("name")
+      .setDescription("Our beloved O.A.F.'s new name")
+      .setRequired(true),
+  );
+
+export async function execute(interaction: ChatInputCommandInteraction) {
+  const member = interaction.member;
+
+  if (!member) {
+    interaction.reply({
+      content: "You have to perform this action from within a Guild.",
+      ephemeral: true,
+    });
+    return;
+  }
+
+  const roleManager = member.roles as GuildMemberRoleManager;
+
+  if (!roleManager.cache.has(config.EXTENDED_TEAM_ROLE_ID)) {
+    interaction.reply({
+      content: "You are not permitted to rename me, your beloved O.A.F.",
+      ephemeral: true,
+    });
+    return;
+  }
+
+  const name = interaction.options.getString("name", true);
+  await discordClient.user?.setUsername(name);
```

Comment:
My only concern is this `user?`. When is it null? Can we error out if it is null?